dot/parachain/candidate-validation/candidate_validation.go

```diff
@@ -211,3 +235,67 @@ func validateFromChainState(runtimeInstance parachainruntime.RuntimeInstance, po
 
 	return &candidateCommitments, persistedValidationData, isValid, nil
 }
+
+// validateFromExhaustive validates a candidate parachain block with provided parameters
+func validateFromExhaustive(validationHost parachainruntime.ValidationHost,
+	persistedValidationData parachaintypes.PersistedValidationData,
+	validationCode parachaintypes.ValidationCode,
+	candidateReceipt parachaintypes.CandidateReceipt, pov parachaintypes.PoV) (
+	*parachainruntime.ValidationResult, error) {
```

Comment:
You are returning an Incorrect return type. It should be `ValidationResultMessage`

Reply:
Updated function to return ValidationResult of parachain type.